pyccel/ast/internals.py

```diff
@@ -414,7 +414,7 @@ def max_precision(objs : list, allow_native : bool = True):
         if ndarray_list:
             return get_final_precision(max(ndarray_list, key=attrgetter('precision')))
         if numpy_v1:
-            return max((get_final_precision(o) for o in objs), default = -1)
+            return max((get_final_precision(o) for o in objs))
```

Comment:
Do you still need the double parentheses?
```python
return max(get_final_precision(o) for o in objs)
```
should work.

Reply:
```suggestion
            return max(get_final_precision(o) for o in objs)
```

Reply:
I agree. Fixed [46ee59f](https://github.com/pyccel/pyccel/pull/1994/commits/46ee59ff99baa6274ca57b1b0421a9e4e7c93915)